### DISCUSSION TITLE
SDK Compatibility with react 18

### DISCUSSION THREAD

**Discussion by skasagar** [pegasystems/react-sdk#448] [Q&A] [answered]
Does this SDK supports react 18 and node 18 minimum version ?  we are trying to integrate this SDK with our application which is built on React 16

Pega react SDK(main branch) is still using react 16 version. Pega react SDK have @pega/react-sdk-components module for UI component which uses @material-ui version 4 which is not supported in react 18

any advice ?

**Comment by egolessengineer** [comment]
Integrating or upgrading to a new version of an SDK in a project that relies on older versions of its core technologies (like React or Node.js) can be challenging, but you can try some approaches.

🛠 **Staying with React 16 (for now)**

If upgrading other parts of your application is not immediately necessary, you might consider staying with React 16 until the SDK offers official support for React 18. 

🛠 **Forking the SDK**

Forking the SDK and updating its dependencies (React and Material-UI) to versions compatible with React 18 could be a solution. This approach allows you to control the upgrade process and make necessary adjustments for compatibility. However, this can introduce a maintenance burden, as you'll need to merge upstream changes from the original SDK into your fork regularly.

🛠 **Shim or Polyfill**

In some cases, you might be able to use shims or polyfills to bridge compatibility gaps between React 16 and React 18 within your project.

🛠 **Contributing to the SDK**

If you have the capability, contributing upgrades or compatibility fixes to the Pega React SDK directly could benefit not only your project but the broader user base of the SDK. This might involve updating the React version, replacing @material-ui version 4 with the newer @mui/material (the successor to Material-UI), and ensuring compatibility with React 18.

Cheers.😊

**Reply by timjmartel** [reply, marked as answer]
Planning to move to React 18 in release SDK-R 24.1.10.  Also, exploring moving to MUI 5 in same release.  TBD. Release will be available in Q4/2024.
